From 77c198905ca5218db9cb78569d5a53e2dbd0a8eb Mon Sep 17 00:00:00 2001
From: Konrad Rzeszutek Wilk <konrad.wilk@oracle.com>
Date: Mon, 28 Aug 2017 13:59:12 -0400
Subject: [PATCH 2/2] Use grub-file to figure out whether multiboot2 should be
 used for Xen.gz

The multiboot2 is much more preferable than multiboot. Especiall
if booting under EFI where multiboot does not have the functionality
to pass ImageHandler.

Signed-off-by: Konrad Rzeszutek Wilk <konrad.wilk@oracle.com>
---
v2: Rebase on top of  d33045ce7ffcb7c1e4a60c14d5ca64b36e3c5abe
---
 util/grub.d/20_linux_xen.in | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/util/grub.d/20_linux_xen.in b/util/grub.d/20_linux_xen.in
index 083bcef..29e015b 100644
--- a/util/grub.d/20_linux_xen.in
+++ b/util/grub.d/20_linux_xen.in
@@ -212,6 +212,10 @@ while [ "x${xen_list}" != "x" ] ; do
     else
 	xen_loader="multiboot"
 	module_loader="module"
+	if ($grub_file --is-x86-multiboot2 $current_xen); then
+		xen_loader="multiboot2"
+		module_loader="module2"
+	fi
     fi
     while [ "x$list" != "x" ] ; do
 	linux=`version_find_latest $list`
-- 
2.1.4

